test/avocado/machineslib.py

```diff
@@ -0,0 +1,166 @@
+import os
+import time
+from subprocess import CalledProcessError
+from testlib_avocado.seleniumlib import SeleniumTest, clickable
+
+SPICE_XML = """
+    <video>
+      <model type='vga' heads='1' primary='yes'/>
+      <alias name='video0'/>
+    </video>
+    <graphics type='spice' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+      <image compression='off'/>
+    </graphics>
+"""
+
+VNC_XML = """
+    <video>
+      <model type='vga' heads='1' primary='yes'/>
+      <alias name='video0'/>
+    </video>
+    <graphics type='vnc' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+    </graphics>
+"""
+
+CONSOLE_XML = """
+    <console type='file'>
+      <target type='serial' port='0'/>
+      <source path='{log}'/>
+    </console>
+"""
+
+PTYCONSOLE_XML = """
+    <serial type='pty'>
+      <source path='/dev/pts/3'/>
+      <target port='0'/>
+      <alias name='serial0'/>
+    </serial>
+    <console type='pty' tty='/dev/pts/3'>
+      <source path='/dev/pts/3'/>
+      <target type='serial' port='0'/>
+      <alias name='serial0'/>
+    </console>
+"""
+
+DOMAIN_XML = """
+<domain type='qemu'>
+  <name>{name}</name>
+  <vcpu>1</vcpu>
+  <os>
+    <type arch='x86_64'>hvm</type>
+    <boot dev='hd'/>
+    <boot dev='network'/>
+  </os>
+  <memory unit='MiB'>256</memory>
+  <currentMemory unit='MiB'>256</currentMemory>
+  <features>
+    <acpi/>
+  </features>
+  <cpu mode='host-model'>
+    <model fallback='forbid'/>
+  </cpu>
+  <devices>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='qcow2'/>
+      <source file='{image}'/>
+      <target dev='hda' bus='ide'/>
+    </disk>
+    <controller type='scsi' model='virtio-scsi' index='0' id='hot'/>
+    <interface type='network'>
+      <source network='default' bridge='virbr0'/>
+      <target dev='vnet0'/>
+    </interface>
+    {console}
+    {graphics}
+  </devices>
+</domain>
+"""
+
+
+class MachinesLib(SeleniumTest):
+    """
+    :avocado: disable
+    """
+
+    def create_vm(self, name, graphics='spice', ptyconsole=False, state='running', wait=False):
+        self.virshvm = name
+
+        img = "/var/lib/libvirt/images/cirros.qcow2"
```

Comment:
path ``/var/lib/libvirt/images`` of this could be defined as ``contant``  eg. in top level like ``DOMAIN_XML`` because it is used on more places, and it is connected with storage pool.

Reply:
Agree